src/Umbraco.Web.Common/DependencyInjection/UmbracoBuilderExtensions.cs

```diff
@@ -146,6 +147,10 @@ public static IUmbracoBuilder AddUmbracoCore(this IUmbracoBuilder builder)
         builder.Services.TryAddEnumerable(ServiceDescriptor
             .Singleton<IDatabaseProviderMetadata, CustomConnectionStringDatabaseProviderMetadata>());
 
+        builder.Services.AddSingleton<WebRootFileProviderFactory>();
+        builder.Services.AddSingleton<IManifestFileProviderFactory>(factory => factory.GetRequiredService<WebRootFileProviderFactory>());
+        builder.Services.AddSingleton<IGridEditorsConfigFileProviderFactory>(factory => factory.GetRequiredService<WebRootFileProviderFactory>());
```

Comment:
```suggestion
        builder.Services.AddSingleton<IManifestFileProviderFactory, WebRootFileProviderFactory>();
        builder.Services.AddSingleton<IGridEditorsConfigFileProviderFactory, WebRootFileProviderFactory>();
```

Great job creating specialised interfaces for the individual file provide use cases - love the forward thinking here 👍 

Can't we register them as direct implementations instead of registering `WebRootFileProviderFactory` as a singleton and then using a factory to register the specialised interfaces?